src/storage/copier/table_copy_utils.cpp

```diff
@@ -11,52 +11,32 @@ using namespace kuzu::common;
 namespace kuzu {
 namespace storage {
 
-TableCopyExecutor::TableCopyExecutor(CopyDescription& copyDescription, std::string outputDirectory,
-    TaskScheduler& taskScheduler, Catalog& catalog, common::table_id_t tableID,
-    TablesStatistics* tablesStatistics)
-    : logger{LoggerUtils::getLogger(LoggerConstants::LoggerEnum::LOADER)},
-      copyDescription{copyDescription}, outputDirectory{std::move(outputDirectory)},
-      taskScheduler{taskScheduler}, catalog{catalog}, numRows{0},
-      tableSchema{catalog.getReadOnlyVersion()->getTableSchema(tableID)}, tablesStatistics{
-                                                                              tablesStatistics} {}
-
-uint64_t TableCopyExecutor::copy(processor::ExecutionContext* executionContext) {
-    logger->info(StringUtils::string_format("Copying {} file to table {}.",
-        CopyDescription::getFileTypeName(copyDescription.fileType), tableSchema->tableName));
-    populateInMemoryStructures(executionContext);
-    updateTableStatistics();
-    saveToFile();
-    logger->info("Done copying file to table {}.", tableSchema->tableName);
-    return numRows;
-}
-
-void TableCopyExecutor::populateInMemoryStructures(processor::ExecutionContext* executionContext) {
-    countNumLines(copyDescription.filePaths);
-    initializeColumnsAndLists();
-    populateColumnsAndLists(executionContext);
-}
-
-void TableCopyExecutor::countNumLines(const std::vector<std::string>& filePaths) {
+offset_t TableCopyUtils::countNumLines(common::CopyDescription& copyDescription,
+    catalog::TableSchema* tableSchema,
+    std::unordered_map<std::string, FileBlockInfo>& fileBlockInfos) {
     switch (copyDescription.fileType) {
     case CopyDescription::FileType::CSV: {
-        countNumLinesCSV(filePaths);
-    } break;
+        return countNumLinesCSV(copyDescription, tableSchema, fileBlockInfos);
+    }
     case CopyDescription::FileType::PARQUET: {
-        countNumLinesParquet(filePaths);
-    } break;
+        return countNumLinesParquet(copyDescription, tableSchema, fileBlockInfos);
+    }
     case CopyDescription::FileType::NPY: {
-        countNumLinesNpy(filePaths);
-    } break;
+        return countNumLinesNpy(copyDescription, tableSchema, fileBlockInfos);
+    }
     default: {
         throw CopyException{StringUtils::string_format("Unrecognized file type: {}.",
             CopyDescription::getFileTypeName(copyDescription.fileType))};
     }
     }
 }
 
-void TableCopyExecutor::countNumLinesCSV(const std::vector<std::string>& filePaths) {
-    numRows = 0;
-    for (auto& filePath : filePaths) {
+offset_t TableCopyUtils::countNumLinesCSV(common::CopyDescription& copyDescription,
```

Comment:
Why do we use offset_t for the type of numTuples?
Shouldn't numTuples be an uint64_t?